src/displayapp/screens/settings/SettingWakeUp.cpp

```diff
@@ -89,19 +89,50 @@ bool SettingWakeUp::Refresh() {
 
 
 void SettingWakeUp::UpdateSelected(lv_obj_t *object, lv_event_t event) {
-  if(event == LV_EVENT_VALUE_CHANGED) {
+  if(event == LV_EVENT_VALUE_CHANGED && !isLocked) {
+    isLocked = true;
+    lv_checkbox_set_checked(cbOption[0], true);
     for(int i = 0; i < optionsTotal; i++) {
       if ( object == cbOption[i] ) {
-        lv_checkbox_set_checked(cbOption[i], true);
-        
-        if ( i == 0 ) { settingsController.setWakeUpMode(Pinetime::Controllers::Settings::WakeUpMode::None); };
-        if ( i == 1 ) { settingsController.setWakeUpMode(Pinetime::Controllers::Settings::WakeUpMode::SingleTap); };
-        if ( i == 2 ) { settingsController.setWakeUpMode(Pinetime::Controllers::Settings::WakeUpMode::DoubleTap); };
-        if ( i == 3 ) { settingsController.setWakeUpMode(Pinetime::Controllers::Settings::WakeUpMode::RaiseWrist); };
-        
-      } else {
-        lv_checkbox_set_checked(cbOption[i], false);
+        auto& currentCheckbox = cbOption[i];
+        auto currentCheckboxState = !lv_checkbox_is_checked(currentCheckbox);
```

Comment:
I think it is cleaner now to take this back to.
```
auto currentCheckboxState = lv_checkbox_is_checked(currentCheckbox);
```
And instead change to
```
if (currentCheckboxState) {
```
the 3 times it is used. Easier to read and makes more sense that currentCheckboxState is the actual checkbox state at that time.

Reply:
good idea, done!